diff --git a/main.cpp b/main.cpp
index d6555a6..cf13bd1 100644
--- a/main.cpp
+++ b/main.cpp
@@ -1,6 +1,7 @@
 #define WLR_USE_UNSTABLE
 
 #include <hyprland/src/Compositor.hpp>
+#include <hyprland/src/desktop/state/FocusState.hpp>
 #include <hyprland/src/managers/KeybindManager.hpp>
 
 #include "globals.hpp"
@@ -18,7 +19,7 @@ void hkCXDGToplevelResConstructor(CXDGToplevelResource* thisptr, SP<CXdgToplevel
     resource->setMove([thisptr](CXdgToplevel* t, wl_resource* seat, uint32_t serial) {
         // treat it as down event, moustButton event is used for up event only
         if (auto window = thisptr->m_window.lock()) {
-            g_pCompositor->focusWindow(window);
+            Desktop::focusState()->rawWindowFocus(window);
             g_pKeybindManager->m_dispatchers["mouse"]("1movewindow");
         }
     });
